### PR TITLE
Update modal.js to use ref callbacks instead of strings

### PR DESCRIPTION
**Pull request by stryju**
using string `ref`s can be dangerous, so it's safer to use callback form :)

https://github.com/facebook/react/issues/1373

### CODE REVIEW

File: components/modal.js [hunk 1]
```diff
@@ -14,8 +14,8 @@ export default class extends React.Component {
 
   render () {
     return (
-      <div className={style(styles.shim)} ref="shim" onClick={(e) => this.dismiss(e)}>
-        <div ref="photoWrap" className={style(styles.photo)}>
+      <div ref={(el) => this._shim = el} className={style(styles.shim)} onClick={(e) => this.dismiss(e)}>
```
Comment (justingreenberg) [inline components/modal.js]: Since this is an assignment, callback function should probably be wrapped in braces to avoid implicit return

Comment (stryju) [inline components/modal.js]: this is outside the scope of this diff (reused the previous one) :)

Comment (impronunciable) [inline components/modal.js]: @justingreenberg you are both right. I'm merging this and then doing the parenthesis wrapping change in another commit. Thank you
